src/iceberg/expression/aggregate.h

```diff
@@ -0,0 +1,230 @@
+
+ protected:
+  CountAggregate(Expression::Operation op, std::shared_ptr<BoundTerm> term)
+      : BoundAggregate(op, std::move(term)) {}
+};
+
+/// \brief COUNT(term) aggregate.
+class ICEBERG_EXPORT CountNonNullAggregate : public CountAggregate {
+ public:
+  static Result<std::shared_ptr<CountNonNullAggregate>> Make(
+      std::shared_ptr<BoundTerm> term);
+
+  std::string ToString() const override;
+  Result<Literal> Evaluate(const StructLike& data) const override;
+  Result<std::unique_ptr<Aggregator>> NewAggregator() const override;
+
+ private:
+  explicit CountNonNullAggregate(std::shared_ptr<BoundTerm> term);
+};
+
+/// \brief COUNT_NULL(term) aggregate.
+class ICEBERG_EXPORT CountNullAggregate : public CountAggregate {
+ public:
+  static Result<std::shared_ptr<CountNullAggregate>> Make(
+      std::shared_ptr<BoundTerm> term);
+
+  std::string ToString() const override;
+  Result<Literal> Evaluate(const StructLike& data) const override;
+  Result<std::unique_ptr<Aggregator>> NewAggregator() const override;
+
+ private:
+  explicit CountNullAggregate(std::shared_ptr<BoundTerm> term);
+};
+
+/// \brief COUNT(*) aggregate.
+class ICEBERG_EXPORT CountStarAggregate : public CountAggregate {
+ public:
+  static Result<std::shared_ptr<CountStarAggregate>> Make();
+
+  std::string ToString() const override;
+  Result<Literal> Evaluate(const StructLike& data) const override;
+  Result<std::unique_ptr<Aggregator>> NewAggregator() const override;
+
+ private:
+  CountStarAggregate();
+};
+
+/// \brief Bound MAX/MIN aggregate.
+class ICEBERG_EXPORT ValueAggregate : public BoundAggregate {
+ public:
+  ValueAggregate(Expression::Operation op, std::shared_ptr<BoundTerm> term);
+
+  Kind kind() const override { return Kind::kValue; }
+
+  std::string ToString() const override;
+  Result<Literal> Evaluate(const StructLike& data) const override;
+  Result<std::unique_ptr<Aggregator>> NewAggregator() const override;
+};
+
+/// \brief Evaluates bound aggregates over StructLike rows.
+class ICEBERG_EXPORT AggregateEvaluator {
+ public:
+  virtual ~AggregateEvaluator() = default;
+
+  /// \brief Create an evaluator for a single bound aggregate.
+  /// \param aggregate The bound aggregate to evaluate across rows.
+  static Result<std::unique_ptr<AggregateEvaluator>> Make(
+      std::shared_ptr<BoundAggregate> aggregate);
+
+  /// \brief Create an evaluator for multiple bound aggregates.
+  /// \param aggregates Aggregates to evaluate in one pass; order is preserved in
+  /// Results().
+  static Result<std::unique_ptr<AggregateEvaluator>> MakeList(
+      std::vector<std::shared_ptr<BoundAggregate>> aggregates);
+
+  /// \brief Update aggregates with a row.
+  virtual Status Update(const StructLike& row) = 0;
+
+  /// \brief Final aggregated value.
+  virtual Result<std::vector<Literal>> Results() const = 0;
+
+  /// \brief Convenience accessor when only one aggregate is evaluated.
+  virtual Result<Literal> ResultLiteral() const = 0;
```

Comment:
```suggestion
  virtual Result<std::vector<Literal>> GetResults() const = 0;

  /// \brief Convenience accessor when only one aggregate is evaluated.
  virtual Result<Literal> GetResult() const = 0;
```

Perhaps this looks more consistent.